pilot/pkg/networking/core/v1alpha3/cluster_builder.go

```diff
@@ -775,6 +776,11 @@ func (cb *ClusterBuilder) buildUpstreamClusterTLSContext(opts *buildClusterOpts,
 				CaCertificatePath: tls.CaCertificates,
 			}
 
+			// Use service accounts in service if TLS settings does not have subject alt names for service entries.
+			if opts.serviceRegistry == string(serviceregistry.External) && len(tls.SubjectAltNames) == 0 {
+				tls.SubjectAltNames = opts.serviceAccounts
```

Comment:
Can you move the tls setting to `buildAutoMtlsSettings`? It belongs there.

Reply:
I do not think it is autoMTLS related. For user specified TLS settings also for service entries if subject alt names are provided there and not in DR - we should handle it. This actually belongs here.

Reply:
Actually，in some case the service accounts are already set there. We may need to rename that function

Reply:
Here it is not service accounts. It is actually subject alt names given in service entry. During conversion it is stored in service https://github.com/istio/istio/blob/master/pilot/pkg/serviceregistry/serviceentry/conversion.go#L220 and we copy it to opts. So opts.serviceAccounts is actually sANs for SEs - which is confusing but not changed in this PR

Reply:
I think both sa and SubjectAltNames are all used to build same field. They should be set when complete tls.

Also we have passed  tls etting to `buildUpstreamClusterTLSContext`, if we set tls again in some case, this is very coupling with `buildAutoMtlsSettings`

Reply:
`buildAutoMtlsSettings`  as is meant today only deals with auto mTLS cases. If we have to do what you are saying we need to change that buildMtlsSettings and take care of SIMPLE also. I can try to refactor that later - but IMO it is not related to this PR. 

We do set tls like this - in this method already. 
		if len(tlsContext.Sni) == 0 {
			tlsContext.Sni = c.cluster.Name
		}